src/storage.rs

```diff
@@ -158,6 +168,24 @@ impl TryFrom<capi::StorageProperties> for StorageProperties {
                     .to_owned(),
             )
         };
+        let s3_access_key_id = if value.access_key_id.nbytes == 0 {
+            None
+        } else {
+            Some(
+                unsafe { CStr::from_ptr(value.access_key_id.str_) }
+                    .to_str()?
+                    .to_owned(),
+            )
+        };
+        let s3_secret_access_key = if value.secret_access_key.nbytes == 0 {
+            None
+        } else {
+            Some(
+                unsafe { CStr::from_ptr(value.secret_access_key.str_) }
+                    .to_str()?
+                    .to_owned(),
+            )
+        };
```

Comment:
I haven't used Rust before, so the code may be wrong, but there might be an opportunity for a utility function here to reduce duplication and isolate unsafe operations.

```rs
fn cstr_to_string(ptr: *const i8, length: usize) -> Result<Option<String>, Utf8Error> {
    if length == 0 {
        Ok(None)
    } else {
        unsafe {
            let cstr = CStr::from_ptr(ptr);
            let s = cstr.to_str()?.to_owned();
            Ok(Some(s))
        }
    }
}

let s3_access_key_id = cstr_to_string(value.access_key_id.str_, value.access_key_id.nbytes)?;
let s3_secret_access_key = cstr_to_string(value.secret_access_key.str_, value.secret_access_key.nbytes)?;
```